### PR TITLE
Fix Loop Animations for Non-VRM Sourced Rigs

### PR DESCRIPTION
**Pull request by AidanCaruso**
## References
closes #_insert number here_

### CODE REVIEW

File: packages/engine/src/avatar/functions/updateVRMRetargeting.ts [hunk 1]
```diff
@@ -55,23 +49,10 @@ export const updateVRMRetargeting = (vrm: VRM, deltaTime: number) => {
...
-
-        const parentBoneNode = getComponent(parentEntity, BoneComponent)
-        if (!parentBoneNode) continue
-
-        computeTransformMatrix(parentEntity)
-        parentBoneNode.updateMatrixWorld(true)
-
-        const parentWorldMatrix = parentBoneNode.matrixWorld
-        const localPosition = boneWorldPosition.applyMatrix4(parentWorldMatrix.clone().invert())
+        boneNode.parent!.updateWorldMatrix(true, false)
```
Comment (HexaField) [inline packages/engine/src/avatar/functions/updateVRMRetargeting.ts:53]: why are we reverting from ECS language to threejs?

The two clone() calls are also going to be unnecessary and expensive

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/avatar/functions/updateVRMRetargeting.ts (hunk 1 above)
Comment (AidanCaruso) [reply packages/engine/src/avatar/functions/updateVRMRetargeting.ts]: Removed one of the clone calls, but I can't find a way around cloning the matrix world right now.

Comment (HexaField) [reply packages/engine/src/avatar/functions/updateVRMRetargeting.ts]: module scope
`const mat4 = new Matrix4()`

function scope
`const localPosition = boneWorldPosition.applyMatrix4(mat4.copy(parentWorldMatrix).invert())`

Comment (AidanCaruso) [reply packages/engine/src/avatar/functions/updateVRMRetargeting.ts]: Thanks. Don't know why I didn't realize that 🤦

**Review by HexaField — APPROVED** [review]

**Pull request merged**